NCS.DSS.NotificationsListener/NotificationsListener/Function/NotificationsListenerHttpTrigger.cs

```diff
@@ -1,86 +1,83 @@
+            ILogger<NotificationsListenerHttpTrigger> logger) 
         {
             _httpRequestMessageHelper = httpRequestMessageHelper;
-            _saveNotification = saveNotification;
-            _logger = logger;
+            _cosmosDBService = cosmosDBService;
             _dynamicHelper = dynamicHelper;
+            _logger = logger;
         }
 
-        [Function("Post")]
-        [ProducesResponseType(typeof(Models.Notification), (int)HttpStatusCode.OK)]
+        [Function("POST")]
+        [ProducesResponseType(typeof(Notification), (int)HttpStatusCode.OK)]
         [Response(HttpStatusCode = (int)HttpStatusCode.Created, Description = "Notification Created", ShowSchema = true)]
         [Response(HttpStatusCode = (int)HttpStatusCode.NoContent, Description = "Notification does not exist", ShowSchema = false)]
         [Response(HttpStatusCode = (int)HttpStatusCode.BadRequest, Description = "Request was malformed", ShowSchema = false)]
         [Response(HttpStatusCode = (int)HttpStatusCode.Unauthorized, Description = "API key is unknown or invalid", ShowSchema = false)]
         [Response(HttpStatusCode = (int)HttpStatusCode.Forbidden, Description = "Insufficient access", ShowSchema = false)]
-        [Display(Name = "Post", Description = "Ability to create a new transfer resource.")]
+        [Display(Name = "POST", Description = "Mock DSS Prime notification behaviour")]
         public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "URLEndpoint")] HttpRequest req)
         {
-            _logger.LogInformation("Processing a new notification request.");
-
-            Models.Notification notification;
+            _logger.LogInformation("{FunctionName} has been invoked", typeof(NotificationsListenerHttpTrigger));
 
+            Notification notification;
             try
             {
-                _logger.LogInformation("Attempting to deserialize the notification from the request.");
-                notification = await _httpRequestMessageHelper.GetResourceFromRequest<Models.Notification>(req);
+                _logger.LogInformation("Attempting to deserialize the Notification from the request");
+                notification = await _httpRequestMessageHelper.GetResourceFromRequest<Notification>(req);
+
+                if (notification == null)
+                {
+                    _logger.LogError("Notification object is NULL after deserialization.");
+                    return new UnprocessableEntityResult();
+                }
             }
             catch (JsonException ex)
             {
-                _logger.LogError(ex, "Failed to deserialize the notification.");
+                _logger.LogError(ex, "Failed to deserialize the Notification from the request body. Exception {ErrorMessage}", ex.Message);
                 return new UnprocessableEntityObjectResult(_dynamicHelper.ExcludeProperty(ex, PropertyToExclude));
             }
 
```

Comment:
Would be good to add a success log for the getResourceFromRequest invocation above